README
DTS-Task version 0.01
=====================

First version released. The API is not complete.

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

ENABLING EXTENDED TESTS

Extended tests will connect to a MS SQL Server database to fetch DTS package
information and therefore tests the methods.

Enabling extended test is a good idea: in fact, executing only the simple tests
will not test much thing and mocking DTS objects is not a simple task. If you're going
to extended or modify the DTS API, is a very good idea to enable extended tests.

To enable the extended tests is necessary to:

1 - Save the DTS package sample ("test_all.dts") that comes with the module tarball in the database.
2 - Edit the XML file "test-config.xml". If your server uses trusted method to authenticate 
then the only thing that needs to be edit is the servername. Otherwise, add the 
<user>user</user> and <password>password</password> inside the "credential" tag. Of course, 
do not forget to use valid user and password information. The "test_conf.dtd" should help you a bit if 
you are able to use a XML validator.

DEPENDENCIES

This module requires these other modules to work:

- Carp
- Class::Accessor
- Data::Dumper
- Win32::OLE
- Hash::Util

COPYRIGHT AND LICENCE

Copyright (C) 2006 by Alceu Rodrigues de Freitas Junior

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.8 or,
at your option, any later version of Perl 5 you may have available.




CHANGELOG
2008-09-24  alceu.freitas.jr

	* [r83] lib/DTS/Package/Step.pm: add_global_vars does not needs get/set methods

2008-09-23  alceu.freitas.jr

	* [r82] .: updating version to 0.05
	* [r81] META.yml: updating version to 0.05
	* [r80] DTS.ppd: updating version to 0.05
	* [r79] CHANGELOG: initial release of CHANGELOG
	* [r78] META.yml: updated version

2008-08-14  alceu.freitas.jr

	* [r77] TODO.txt:
	* [r76] lib/DTS/Task/SendEmail.pm: changed new method to use a simpler way copy the
	  attribute values
	  POD updated
	* [r75] lib/DTS/DateTime.pm: added parameter verification with Params::Validate module
	* [r74] lib/DTS/Package/Step.pm: fixed bug regarding exec_time attribute (double, not
	  datetime)
	  added POD for all missing methods
	* [r73] MANIFEST, t/DateTime.t, t/extended/Task-SendEmail.t: included more test files
	* [r72] lib/DTS.pm: updated POD

2008-08-13  alceu.freitas.jr

	* [r71] examples/dts-ut.pl, examples/test.pl: removed test examples, they will be replace by
	  a more updated example that is implemented together with a web application for DTS unit
	  testing
	* [r70] META.yml: added resources
	* [r69] lib/DTS/Package/Step.pm, lib/DTS/Package/Step/Result.pm, t/extended/StepResult.t,
	  t/StepResult.t:

2008-08-12  alceu.freitas.jr

	* [r68] lib/DTS.pm, lib/DTS/DateTime.pm, lib/DTS/Package.pm, lib/DTS/Package/Step.pm,
	  lib/DTS/Package/Step/Result.pm, t/extended/Assignment-All-Destination.t,
	  t/extended/Package.t: added extended tests for extended/Assignment-All-Destination.t and
	  extended/Package.t
	  
	  fixed bugs in DTS/DateTime.pm DTS/Package.pm, DTS/Package/Step.pm and
	  DTS/Package/Step/Result.pm
	  
	  updated POD in DTS.pm
	* [r67] lib/DTS/Package/Step.pm, lib/DTS/Package/Step/Result.pm, MANIFEST,
	  t/extended/Package.t, t/Package.t: Changes related to add support for executing steps and
	  checking their results.
	* [r66] lib/DTS/Package/Step.pm: All changes related to be able to execute a Package steps
	  DTS::DateTime will be a generic DateTime object constructed with a Win32::OLE::Variant
	  object.
	  THIS CODE IS INCOMPLETE!

2008-08-11  alceu.freitas.jr

	* [r65] TODO.txt: Changes related to add support for executing steps and checking their
	  results.
	* [r64] t/DTS.t: a simple test file for superclass DTS.pm
	* [r63] lib/DTS/Package/Step.pm: Changes related to add support for executing steps and
	  checking their results.
	* [r62] lib/DTS/DateTime.pm, lib/DTS/Package.pm, lib/DTS/Package/Step,
	  lib/DTS/Package/Step.pm, lib/DTS/Package/Step/Result.pm, lib/DTS/Task/DynamicProperty.pm,
	  MANIFEST: Changes related to add support for executing steps and checking their results.
	* [r61] lib/DTS.pm, lib/DTS/DateTime.pm, lib/DTS/Package, lib/DTS/Package.pm,
	  lib/DTS/Package/Step.pm, MANIFEST: All changes related to be able to execute a Package
	  steps
	  DTS::DateTime will be a generic DateTime object constructed with a Win32::OLE::Variant
	  object.
	  THIS CODE IS INCOMPLETE!

2008-07-31  alceu.freitas.jr

	* [r60] .:
	* [r59] DTS.ppd: modified for release 0.04
	* [r58] Makefile.PL: added dist parameters

2008-07-29  alceu.freitas.jr

	* [r57] lib/DTS.pm: changed to version 0.04
	* [r56] t/extended/Assignment-All-Destination.t, t/extended/Assignment-Constant.t,
	  t/extended/Assignment-DataFile.t, t/extended/Assignment-EnvVar.t,
	  t/extended/Assignment-GlobalVar.t, t/extended/Assignment-INI.t,
	  t/extended/Assignment-Query.t: removed warnings about subroutines redefined

2008-07-28  alceu.freitas.jr

	* [r55] t/extended/Assignment-All-Destination.t: fixed bug
	* [r54] t/extended/Assignment.t: fixed bugs
	* [r53] t/extended/Assignment-GlobalVar.t: fixed bugs
	* [r52] t/extended/Assignment-EnvVar.t, t/extended/Assignment-INI.t,
	  t/extended/Assignment-Query.t: fixed bugs
	* [r51] t/extended/Assignment-All-Destination.t, t/extended/Assignment-Constant.t,
	  t/extended/Assignment-DataFile.t, t/extended/Assignment-EnvVar.t,
	  t/extended/Assignment-GlobalVar.t: several fixes in the tests files
	* [r50] t/Assignment-Destination.t: fixed wrong number of tests

2008-07-26  alceu.freitas.jr

	* [r49] t/Assignment-Destination.t:

2008-07-25  alceu.freitas.jr

	* [r48] ., lib/DTS/Assignment.pm, lib/DTS/Assignment/Constant.pm,
	  lib/DTS/Assignment/DataFile.pm, lib/DTS/Assignment/Destination.pm,
	  t/extended/Assignment-DataFile.t, test-config.xml: fixes in .pm files.
	  test file changed to use the new methods available instead of using Win32::OLE functions

2008-07-24  alceu.freitas.jr

	* [r47] t/extended/Assignment-Constant.t: fixed bugs in comparing destination structure
	* [r46] lib/DTS/Assignment.pm:
	* [r45] lib/DTS/Assignment.pm: fixed bug regarding get_type and get_type_name
	* [r44] lib/DTS/Application.pm: Fixed bug regarding new method, which was not providing a
	  reference to a DTS.Application object (_sibling)
	* [r43] Makefile.PL: moved Devel::AssertOS to DTS.pm
	* [r42] lib/DTS.pm, lib/DTS/Task.pm:
	* [r41] t/Assignment-Destination.t, t/AssignmentTypes.t, t/Task.t, t/Task-DataPump.t,
	  t/Task-DynamicProperty.t, t/Task-ExecutePackage.t, t/Task-SendEmail.t: several fixes in
	  the tests scripts
	* [r40] t/Destination-Connection.t: fixed test from ok() to live_ok() since set_string does
	  not returns anything
	* [r39] lib/DTS/Assignment/Destination.pm: changed new() and set_string methods
	* [r38] t/Assignment-Destination.t: test file related to release 0.3
	* [r37] examples/benchmark.pl, examples/list.pl, examples/modify.pl, examples/modify.xml,
	  examples/README.txt, examples/save_pkgs.pl, Makefile.PL: adding more examples related to
	  release 0.3
	* [r36] lib/DTS/Assignment/Destination, lib/DTS/Assignment/Destination.pm,
	  lib/DTS/Assignment/Destination/Connection.pm, lib/DTS/Assignment/Destination/GlobalVar.pm,
	  lib/DTS/Assignment/Destination/Package.pm, lib/DTS/Assignment/Destination/Task.pm,

Makefile.PL
use 5.008008;
use ExtUtils::MakeMaker qw(WriteMakefile prompt);

my %config = (
    NAME         => 'DTS',
    VERSION_FROM => 'lib/DTS.pm',    # finds $VERSION
    PREREQ_PM    => {
        'Class::Accessor'  => 0.25,
        'Hash::Util'       => 0.05,
        'Win32::OLE'       => 0.1704,
        'Carp'             => 1.04,
        'DateTime'         => 0.35,
        'Class::Publisher' => 0.2,
        'Devel::AssertOS'  => 0
    },
    META_MERGE => {
        resources => {
            license  => 'http://www.gnu.org/licenses/old-licenses/gpl-2.0.html',
            homepage => 'http://code.google.com/p/perldts/',
            bugtracker => 'http://code.google.com/p/perldts/issues/list',
            repository => 'http://code.google.com/p/perldts/source/browse/'
        },
    },
    LICENSE => 'GNU General Public License (GPL)',

    # :WORKAROUND:14/10/2008:arfreitas: ExtUtils::MakeMaker version available
    # for ActivePerl does not support META_MERGE tags. Until an upgrade
    # is available, I'll disable META.yml generation, writing it by hand
    NO_META => 1,
    dist    => { SUFFIX => '.gz', COMPRESS => 'gzip --best' },
    (
        $] >= 5.005
        ?    ## Add these new keywords supported since 5.005
          (
            ABSTRACT_FROM => 'lib/DTS.pm',    # retrieve abstract from module
            AUTHOR => 'Alceu Rodrigues de Freitas Junior <arfreitas@cpan.org>'
          )
        : ()
    ),
);

my @patterns = qw( t/*.t );

print
"You can enable extended tests if the environment was prior configured. See README for more information\n";
my $answer = prompt( 'Do you want to enable extended tests?', 'no' );

if ( $answer =~ /^y/i ) {

    print "Running extended tests.\n";
    push @patterns, 't/extended/*.t';

}
else {

    print "Skipping extended tests.\n";

}

$config{test} = { TESTS => join ' ', map { glob } @patterns };
WriteMakefile(%config);

